### ISSUE TITLE
Pulumi.Aws.Eks.ClusterArgs.VpcConfig.SecurityGroupsIds not manage "Cluster security group" but add "Additional security groups"

### ISSUE THREAD

**Issue by guitarrapc** [pulumi/pulumi-aws#805]
language: dotnet 

## TL;DR

`SecurityGroupIds` should manage EKS Cluster's `Cluster security group`, but security groups are add to `Additional security groups`.

## Summary

`Pulumi.Aws.Eks.ClusterArgs` class will manage how eks cluster is configured and `VpcConfig.SecurityGroupIds` is the property to assign user security group.

However, even if I assign security group ids to `SecurityGroupIds` but they all add to `Additional security groups` and `Cluster security group` will be automatically generated via EKS itself.

It result `pulumi destory` will stack with VPC because of EKS auto generated sg reference itself id in Ingress rule. (Manual remove ingress operation is required.)

## Reproduce

```csharp
new Pulumi.Aws.Eks.Cluster($"sandbox-cluster", new Pulumi.Aws.Eks.ClusterArgs
{
    RoleArn = "arn:input_your_role_arn",
    Version = "1.14",
    VpcConfig = new ClusterVpcConfigArgs
    {
        SecurityGroupIds = new [] { "sg-0a513395f5cf69158" },
        SubnetIds = new [] { "subnet-0777dd6b37013057a", "subnet-0d8e7a2afbcba86ae" },
    },
});
```

## Expected behavior

`sg-0a513395f5cf69158` would reference with `Cluster security group`.

## Actual behavior

`sg-0a513395f5cf69158` would reference with `Additional security groups`.
`Cluster security group`'s `sg-0d475b96dc7b8a54e` is automatically generated via EKS.

![image](https://user-images.githubusercontent.com/3856350/69602540-4d1c7380-105b-11ea-9b55-2f68a0a66c21.png)

![image](https://user-images.githubusercontent.com/3856350/69602614-966cc300-105b-11ea-9a7a-f791ad46ff80.png)

## Terraform

Terraform will assign user security group to `Cluster security group`, so current pulumi behavior is differ from original.

```terraform.tf
resource "aws_eks_cluster" "main" {
  name     = "sandbox-cluster"
  role_arn = "arn:input_your_role_arn"
  version  = "1.14"
  vpc_config {
    subnet_ids         = ["subnet-0777dd6b37013057a", "subnet-0d8e7a2afbcba86ae"]
    security_group_ids = ["sg-0a513395f5cf69158"]
  }
}
```

**Comment by guitarrapc** [comment]
## Version

```shell
$ pulumi version
v1.6.0
$ pulumi plugin ls
NAME  KIND      VERSION  SIZE    INSTALLED    LAST USED
aws   resource  1.10.0   185 MB  3 hours ago  3 hours ago
```

NuGet

```
    <PackageReference Include="Pulumi.Aws" Version="1.10.0-preview" />
    <PackageReference Include="Pulumi" Version="1.6.0-preview" />
```

**Comment by stack72** [comment]
Hi @guitarrapc 

Apologies that you are facing this issue. So I have been investigating this issue today. Unfortunately, this isn't actually something Pulumi specific. So I was using the following code:

```
var vpc = new Vpc("my-vpc", new VpcArgs
            {
                CidrBlock = "192.168.0.0/16",
            });

            var subnet1 = new Subnet($"my-public-subnet-1", new SubnetArgs
            {
                VpcId = vpc.Id,
                CidrBlock = "192.168.0.0/17",
                AvailabilityZone = "us-west-2a"
            }); 
            
            var subnet2 = new Subnet($"my-public-subnet-2", new SubnetArgs
            {
                VpcId = vpc.Id,
                CidrBlock = "192.168.128.0/17",
                AvailabilityZone = "us-west-2b"
            }); 
            
            var sg = new SecurityGroup("my-sg", new SecurityGroupArgs
            {
                VpcId = vpc.Id,
                Ingress = new List<SecurityGroupIngressArgs>
                {
                    new SecurityGroupIngressArgs
                    {
                        Protocol = "tcp",
                        FromPort = 80,
                        ToPort = 80,
                        CidrBlocks = new List<string>{
                            "0.0.0.0/0"
                        }
                    }
                }
            });

            var role = new Role("my-role", new RoleArgs
            {
                AssumeRolePolicy = @"
{
  ""Version"": ""2012-10-17"",
  ""Statement"": [
    {
      ""Effect"": ""Allow"",
      ""Principal"": {
        ""Service"": ""eks.amazonaws.com""
      },
      ""Action"": ""sts:AssumeRole""
    }
  ]
}
".Trim(),
            });
            
            var policies = new List<string>
            {
                "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
                "arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
            };

            var counter = 0;
            foreach (var policy in policies)
            {
                new RolePolicyAttachment($"policy-attachment-{counter}", new RolePolicyAttachmentArgs
                {
                    PolicyArn = policy, 
                    Role = role.Arn
                });
                counter++;
            }

            var cluster = new Cluster("my-cluster", new ClusterArgs
            {
                RoleArn = role.Arn,
                VpcConfig = new ClusterVpcConfigArgs
                {
                    SecurityGroupIds = new InputList<string>
                    {
                        sg.Id
                    },
                    SubnetIds = new InputList<string>
                    {
                        subnet1.Id,
                        subnet2.Id
                    },
                    VpcId = vpc.Id,
                }
            });
```

As you can see, I am passing a security group to the EKS cluster. When the cluster was provisioning, I could see that the correct security group was attached to the cluster. When the cluster had successfully provisioned, I could see that EKS itself had created a new security group and it was the group attached to the cluster. The security group I had passed had been swapped to be an additional security group. I was puzzled by this and found this article https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html

TL;DR from Kubernetes 1.14 on EKS, AWS creates a security group on your behalf that allows the cluster to talk to the managed node group

I hope this clears up the issue for you - I don't believe this is a pulumi bug and think we can close this off

Thanks

Paul

**Comment by guitarrapc** [comment]
thank you stack72, I was missing a doc as it. (this note is only describe in English Doc, though.)

As you quote, and more specificly `Kubernetes 1.14  and eks.3 (or higher)` would change the behaviour.

> Kubernetes version 1.14 and platform version eks.3

I've confirmed eks.5 cluster has this behaviour and eks.2 isn't.
I believe this is by AWS design, not Pulumi issue.

Let's close the case!

**Comment by stack72** [comment]
Thanks @guitarrapc 

Let me know if you have any other issues

Paul
